src/core/algorithms/md/hymd/preprocessing/similarity_measure/distance_similarity_measure.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include "algorithms/md/hymd/lowest_bound.h"
+#include "algorithms/md/hymd/preprocessing/ccv_id_pickers/index_uniform.h"
+#include "algorithms/md/hymd/preprocessing/similarity_measure/similarity_measure.h"
+#include "model/types/double_type.h"
+
+namespace algos::hymd::preprocessing::similarity_measure {
+using DistanceFunction = std::function<size_t(std::byte const*, std::byte const*)>;
+
+class DistanceSimilarityMeasure : public SimilarityMeasure {
+private:
+    DistanceFunction const compute_distance_;
+    model::md::DecisionBoundary min_sim_;
+    util::WorkerThreadPool* const pool_;
+    ccv_id_pickers::IndexUniform picker_{size_limit_};
+    std::size_t const size_limit_;
```

Comment:
```suggestion
    std::size_t const size_limit_;
    ccv_id_pickers::IndexUniform picker_{size_limit_};
```
Earlier members are initialized first